### PR TITLE
Added alias for F in basemap

### LINKED ISSUE #1893: Missing alias for option -F in basemap
**Description of the problem**

I am trying to add a filled background behind my mapscale, similar to the examples in the GMT documentation here: https://docs.generic-mapping-tools.org/dev/cookbook/features.html#placing-map-scales. From what I understand, when adding the map_scale, I should be able to simultaneously add a frame option specifying the fill parameters, like in my example code below, using the options from the GMT documentation. The code below runs when 'frame' isn't specified, but of course no background is generated behind the map_scale. I would be very appreciative if someone could point me in the right direction or let me know if this is in fact a bug. Thanks!

**Full code that generated the error**

```python
import pygmt

fig = pygmt.Figure()
fig.coast(
    # Set the x-range from 10E to 20E and the y-range to 35N to 45N
    region=[10, 20, 35, 45],
    projection="M12c",
    land="lightgray",
    water="white",
    borders="1/0.5p",
    shorelines="1/0.5p",
    frame="ag",
)
fig.basemap(map_scale='jBR+c36N+w100k+l+f', frame='+p1p+i+gwhite+c0.1i')
fig.show()
```

**Full error message**

```
pygmt.exceptions.GMTCLibError: Module 'basemap' failed with status code 72:
basemap [ERROR]: Option -B: Found frame setting modifiers (+b|g|i|n|o|t|w|x|y|z) mixed with axes settings!
basemap [ERROR]: Option -B parsing failure. Correct syntax:
basemap [ERROR]: Offending option -B+p1p+i+gwhite+c0.1i
```

**System information**

Please paste the output of `python -c "import pygmt; pygmt.show_versions()"`:

```
PyGMT information:
  version: v0.6.1
System information:
  python: 3.9.12 | packaged by conda-forge | (main, Mar 24 2022, 23:24:38)  [Clang 12.0.1 ]
  executable: /opt/homebrew/Caskroom/miniforge/base/bin/python
  machine: macOS-12.3.1-arm64-arm-64bit
Dependency information:
  numpy: 1.22.3
  pandas: 1.4.2
  xarray: 2022.3.0
  netCDF4: 1.5.7
  packaging: 21.3
  geopandas: 0.10.2
  ghostscript: 9.56.1
  gmt: 6.3.0
GMT library information:
  binary dir: /opt/homebrew/Caskroom/miniforge/base/bin
  cores: 10
  grid layout: rows
  library path: /opt/homebrew/Cellar/gmt/6.3.0_2/lib/libgmt.dylib
  padding: 2
  plugin dir: /opt/homebrew/Cellar/gmt/6.3.0_2/lib/gmt/plugins
  share dir: /opt/homebrew/Cellar/gmt/6.3.0_2/share/gmt
  version: 6.3.0
```

### PR DESCRIPTION
**Pull request by jackbeagley**
**Description of proposed changes**

Adds an alias for "box" so that backgrounds and frames for map_scales can be generated more easily.

Fixes #1893 

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [ ] If wrapping a new module, open a 'Wrap new GMT module' issue and submit reasonably-sized PRs.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Comment by michaelgrund** [comment]
Thanks for opening this PR @jackbeagley. Besides adding an alias for ``-F`` could also provide the corresponding documentation part in the docstring? You can orientate yourself on the already existing ones for basemap.

**Comment by seisman** [comment]
> Besides adding an alias for `-F` could you please also provide the corresponding documentation part in the docstring? You can orientate yourself on the already existing ones for basemap.

@jackbeagley FYI, the same `box` parameter has been aliased in the `colorbar` function, so you can simply copy the `box` docstring from `colorbar` to `basemap`:

https://github.com/GenericMappingTools/pygmt/blob/555cd310b0afbb79330df02a5679ccf5358f5343/pygmt/src/colorbar.py#L71

**Comment by jackbeagley** [comment]
@seisman , awesome, I have just copied the description from `colorbar` verbatim, as it seemed very well written and as you say - it is the same scale, so I don't think there is anything additional that needs to be added?

**Comment by jackbeagley** [comment]
Does a test need to be written for this too?

File: pygmt/src/basemap.py [hunk 1]
```diff
@@ -61,6 +62,23 @@ def basemap(self, **kwargs):
         [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
         **+w**\ *length*.
         Draws a simple map scale centered on the reference point specified.
+    box : bool or str
+        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
+        [**+p**\ [*pen*]][**+r**\ [*radius*]][**+s**\ [[*dx*/*dy*/][*shade*]]].
+        If set to ``True``, draws a rectangular border around the color scale.
```
Comment (seisman) [inline pygmt/src/basemap.py:68]:
```suggestion
        If set to ``True``, draws a rectangular border around the map scale.
```

**Comment by seisman** [comment]
> Does a test need to be written for this too?

I don't think a test is needed for such an alias.

**Review by michaelgrund — COMMENTED** [review]
I think we have to consider the different types of map objects that can be manipulated such as  map inset, map scale, or map rose.

**Comment by jackbeagley** [comment]
> I think we have to consider the different types of map objects that can be manipulated such as map inset, map scale, or map rose.

Hey @michaelgrund, is that what you had in mind?

File: pygmt/src/basemap.py [hunk 2]
```diff
@@ -61,6 +62,24 @@ def basemap(self, **kwargs):
...
+        xgap/ygap, or lgap/rgap/bgap/tgap where these items are uniform,
+        separate in x- and y-direction, or individual side spacings between
+        scale and border. Append **+i** to draw a secondary, inner border as
+        well. We use a uniform gap between borders of 2p and the
+        :gmt-term:`MAP_DEFAULTS_PEN` unless other values are specified. Append
+        **+r** to draw rounded rectangular borders instead, with a 6p corner
+        radius. You can override this radius by appending another value.
+        Finally, append **+s** to draw an offset background shaded region.
+        Here, *dx/dy* indicates the shift relative to the foreground frame
+        [4p/-4p] and shade sets the fill style to use for shading
```
Comment (weiji14) [inline pygmt/src/basemap.py:81]: One quick suggestion here
```suggestion
        [Default is 4p/-4p] and shade sets the fill style to use for shading
```

File: pygmt/src/basemap.py [hunk 3]
```diff
@@ -61,6 +62,24 @@ def basemap(self, **kwargs):
         [**g**\|\ **j**\|\ **J**\|\ **n**\|\ **x**]\ *refpoint*\
         **+w**\ *length*.
         Draws a simple map scale centered on the reference point specified.
+    box : bool or str
+        [**+c**\ *clearances*][**+g**\ *fill*][**+i**\ [[*gap*/]\ *pen*]]\
+        [**+p**\ [*pen*]][**+r**\ [*radius*]][**+s**\ [[*dx*/*dy*/][*shade*]]].
+        If set to ``True``, draws a rectangular border around the map inset,
+        map scale or rose. Alternatively, specify a different pen with
```
Comment (seisman) [inline pygmt/src/basemap.py:69]: `map inset` is no longer available in the modern-mode "basemap" module. Users who wants to use `inset` should use the `inset` module instead.
```suggestion
        If set to ``True``, draws a rectangular border around the 
        map scale or rose. Alternatively, specify a different pen with
```

**Review by seisman — APPROVED** [review]

**Review by michaelgrund — APPROVED** [review]

**Pull request merged**